CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+## Unreleased
+
+### Added
+
+- Add error details on failure to checkout worker from pool.
+
+### Changed
+
+- `wait_for` and `evaluate` options can now be combined, as the `wait_for` script is appended to the user-given `evaluate` expression.
```

Comment:
Can you extract this into a separate PR ? It is already hard to review without having 2 in one :D

Reply:
Not really, I used this specifically to make the test work in the way you suggested (using a script that delays the `ready-to-print` thing)